src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/naxxramas/boss_gothik.cpp

```diff
@@ -286,7 +286,7 @@ struct boss_gothikAI : public ScriptedAI
                     m_teleportTimer -= diff;
 
                 // Second time that Gothik teleports back from dead side to living side: open the central gate
-                if (m_teleportCount >= 4)
+                if (m_teleportCount >= 4 || m_creature->GetHealthPercent() <= 30)
```

Comment:
eh, I guess it should've been `<= 30.f` to make it more clear that it's a floating point comparison...